src/ape/api/providers.py

```diff
@@ -354,6 +354,13 @@ def decode(cls, data: Dict) -> "BlockAPI":
         """
 
 
+def not_implemented(fn):
```

Comment:
Should we move this to `utils`?

Reply:
Yes, was thinking this. Maybe call it `raises_not_implemented`